frontend/packages/erd-core/src/features/erd/components/ERDContent/components/TableNode/TableColumnList/TableColumn/TableColumn.tsx

```diff
@@ -12,16 +12,33 @@ type TableColumnProps = {
   handleId: string
   isSource: boolean
   targetCardinality?: CardinalityType | undefined
+  isHovered?: boolean
+  isSelectedTable?: boolean
+  releatedColumns: string[]
 }
 
 export const TableColumn: FC<TableColumnProps> = ({
   column,
   handleId,
   isSource,
   targetCardinality,
+  isHovered,
+  isSelectedTable,
+  releatedColumns,
 }) => {
+  const isRelated = releatedColumns.includes(column.name)
+  const isSelectedOrHovered = (isSelectedTable || isHovered) && isRelated
+  const isHoveredUnselected = isHovered && !isRelated
```

Comment:
Expressions that are possible with CSS should be implemented with CSS.
Liam needs to display a large number of TableNodes, so performance is critical.